devlib/target.py

```diff
@@ -2893,6 +2928,249 @@ def _resolve_paths(self):
+            will be used as target's kernel.
+
+        * ``arch``: Architecture type. Defaults to ``aarch64``.
+
+        * ``cpu_types``: List of CPU ids for QEMU. The list only contains ``cortex-a72`` by
+            default. This parameter is valid for Arm architectures only.
+
+        * ``initrd_image``: This points to the location of initrd image (e.g.,
+            ``rootfs.cpio.xz``) which will be used as target's root filesystem if kernel
+            does not include one already.
+
+        * ``mem_size``: Size of guest memory in MiB.
+
+        * ``num_cores``: Number of CPU cores. Guest will have ``2`` cores by default.
+
+        * ``num_threads``: Number of CPU threads. Set to ``2`` by defaults.
+
+        * ``cmdline``: Kernel command line parameter. It only specifies console device in
+            default (i.e., ``console=ttyAMA0``) which is valid for Arm architectures.
+            May be changed to ``ttyS0`` for x86 platforms.
+
+        * ``enable_kvm``: Specifies if KVM will be used as accelerator in QEMU or not.
+            Enabled by default if host architecture matches with target's for improving
+            QEMU performance.
+    :type qemu_params: Dict
+
+    :param connection_settings: he dictionary which stores connection settings
+        of :attr:`Target.connection_settings`, defaults to None.
+    :type connection_settings: Dict, optional
+
+    :param make_target: Lambda function for creating :class:`Target` based
+        object, defaults to :func:`lambda **kwargs: LinuxTarget(**kwargs)`.
+    :type make_target: func, optional
+
+    :Variable positional arguments: Forwarded to :class:`TargetRunner`.
+
+    :raises FileNotFoundError: if QEMU executable, kernel or initrd image cannot be found.
+    """
+
+    def __init__(self,
+                 qemu_params,
+                 connection_settings=None,
+                 # pylint: disable=unnecessary-lambda
+                 make_target=lambda **kwargs: LinuxTarget(**kwargs),
+                 **args):
+        self.connection_settings = {
+            'host': '127.0.0.1',
+            'port': 8022,
+            'username': 'root',
+            'password': 'root',
+            'strict_host_check': False,
+        }
+
+        # Update default connection settings with :param:`connection_settings` (if exists).
+        if connection_settings is not None:
+            self.connection_settings = { **self.connection_settings, **connection_settings }
+
+        qemu_default_args = {
+            'kernel_image': '',
+            'arch': 'aarch64',
+            'cpu_type': 'cortex-a72',
+            'initrd_image': '',
+            'mem_size': 512,
+            'num_cores': 2,
+            'num_threads': 2,
+            'cmdline': 'console=ttyAMA0',
+            'enable_kvm': True,
+        }
+
+        # Update default QEMU parameters with :param:`qemu_params`.
+        qemu_default_args.update(
```

Comment:
nit: Would making `qemu_default_args` to just be named `qemu_args` be a bit clearer in the following code that it is taking in account the user provided information as well?

Reply:
Done.